spec/rails_state_machine/state_machine/event_spec.rb

```diff
@@ -0,0 +1,103 @@
+describe RailsStateMachine::StateMachine do
+
+  shared_examples 'an invalid record in new state' do
```

Comment:
Several specs in here fail for me because of transition errors, like so:

```
     Failure/Error:
       expect { parcel.ship }.to raise_error(
         RailsStateMachine::Event::TransitionNotFoundError,
         'ship does not transition from empty; defined are [#<struct RailsStateMachine::Event::Transition from=:filled, to=:ship>]'
       )
     
       expected RailsStateMachine::Event::TransitionNotFoundError with "ship does not transition from empty; defined are [#<struct RailsStateMachine::Event::Transition from=:filled, to=:ship>]", got #<RailsStateMachine::Event::TransitionNotFoundError: ship does not transition from empty; defined are [#<struct RailsStateMachine::Event::Transition from=:filled, to=:shipped>]> with backtrace:
...
```


```
rspec ./spec/rails_state_machine/state_machine/event_spec.rb[1:1:1:1:1] # RailsStateMachine::StateMachine when new record behaves like an invalid record in new state #<event_name> does not save the record
rspec ./spec/rails_state_machine/state_machine/event_spec.rb[1:1:1:2:1] # RailsStateMachine::StateMachine when new record behaves like an invalid record in new state #<event_name>! raises an error
rspec ./spec/rails_state_machine/state_machine/event_spec.rb[1:1:2:1:1] # RailsStateMachine::StateMachine when new record behaves like a valid record in new state #<event_name> saves the record
rspec ./spec/rails_state_machine/state_machine/event_spec.rb[1:1:2:2:1] # RailsStateMachine::StateMachine when new record behaves like a valid record in new state #<event_name>! saves the record
rspec ./spec/rails_state_machine/state_machine/event_spec.rb[1:1:3:1:1] # RailsStateMachine::StateMachine when new record behaves like an invalid transition for record #<event_name> raises an error
rspec ./spec/rails_state_machine/state_machine/event_spec.rb[1:1:3:2:1] # RailsStateMachine::StateMachine when new record behaves like an invalid transition for record #<event_name>! raises an error
rspec ./spec/rails_state_machine/state_machine/event_spec.rb[1:2:3:1:1] # RailsStateMachine::StateMachine when persisted record behaves like an invalid transition for record #<event_name> raises an error
rspec ./spec/rails_state_machine/state_machine/event_spec.rb[1:2:3:2:1] # RailsStateMachine::StateMachine when persisted record behaves like an invalid transition for record #<event_name>! raises an error
```